internal/core/metadata/utils.go

```diff
@@ -25,20 +26,20 @@ const (
 
 // Printing function purely for debugging purposes
 // Print the body of a request to the console
-func printBody(r io.ReadCloser) {
+func printBody(r io.ReadCloser, loggingClient logger.LoggingClient) {
 	body, err := ioutil.ReadAll(r)
 	bodyString := string(body)
 
 	if err != nil {
-		LoggingClient.Error(err.Error())
+		loggingClient.Error(err.Error())
 	}
 
-	LoggingClient.Info(bodyString)
+	loggingClient.Info(bodyString)
 }
 
-func checkMaxLimit(limit int) error {
+func checkMaxLimit(limit int, loggingClient logger.LoggingClient) error {
```

Comment:
@michaelestrin I'm wondering if it's more appropriate for `checkMaxLimit` to not log its error and push that responsibility to the caller.

Reply:
Concur.  Further, believe the existing implementation is already logging (via caller's subsequent call to `httpErrorHandler.Handle` when `err != nil`) and thus the call to `loggingClient.Error` below is redundant and can be removed.